fuel-vm/src/tests/crypto.rs

```diff
@@ -82,6 +82,141 @@ fn ecrecover() {
 
     assert!(success);
 }
+
+#[test]
+fn ecrecover_tx_id() {
+    let rng = &mut StdRng::seed_from_u64(2322u64);
+
+    let mut client = MemoryClient::default();
+
+    let gas_price = 0;
+    let gas_limit = 1_000_000;
+    let maturity = Default::default();
+    let height = Default::default();
+    let params = ConsensusParameters::default();
+    let gas_costs = GasCosts::default();
+
+    let secret = SecretKey::random(rng);
+    let public = secret.public_key();
+
+    #[rustfmt::skip]
+    let script = vec![
+        // 0x21 is a address of the singer of the witness
+        op::gtf_args(0x20, 0x00, GTFArgs::ScriptData),
+        op::move_(0x21, 0x20),
+        // 0x22 is a witness - signature
+        op::gtf_args(0x22, 0x00, GTFArgs::WitnessData),
+        // TxId is stored in the first 32 bytes of the memory
+        // Store it into register 0x23
+        op::movi(0x23, 0),
+        // Allocate space for the recovered public key
+        // 0x10 contains the size of the public key = PublicKey::LEN
+        op::movi(0x10, PublicKey::LEN as Immediate18),
+        op::aloc(0x10),
+        op::move_(0x11, RegId::HP),
```

Comment:
Nit: assigning to a register is not needed, `RegId::HP` could be used directly in place of `0x11` after this.